### PR TITLE
Host average rating

### PR DESCRIPTION
**Pull request by vradovic**
This PR changes the host average rating to be fetched from backend.

### CODE REVIEW

File: src/app/host/host-page/host-page.component.html
Comment (MomirMilutinovic) [inline src/app/host/host-page/host-page.component.html]: I suggest leaving this line as is.
```suggestion
        <app-reviews [id]="id || 0" [reportingAllowed]="reportingAllowed" *ngIf="reviews != null" (reviewDeleted)="calculateAverageRating()"></app-reviews>
```

File: src/app/host/host-page/host-page.component.ts
Comment (MomirMilutinovic) [inline src/app/host/host-page/host-page.component.ts:58]: I recommend changing these lines so they fetch the average rating again and not deleting this method.

**Review by MomirMilutinovic — APPROVED** [review]
Looks good. 
There is still a way to refresh the average rating of a host when a review is deleted.
I've left the PR open in case you want to implement that. Feel free to merge this PR if you think that refreshing ratings is not needed.

**Comment by MomirMilutinovic** [comment]
I have rebased this branch on top of the current develop branch. Currently, the host page 404s because the backend for checking eligibility to report a host has not yet been merged (https://github.com/tinylunark/lunark-back/pull/73). 
I recommend delaying the merging of this PR until the aforementioned backend PR has been merged.

File: src/app/http-interceptors/page-not-found.interceptor.ts
Comment (MomirMilutinovic) [inline src/app/http-interceptors/page-not-found.interceptor.ts]: What issues is `PageNotFoundInterceptor` causing?
If it being invoked when acessing `api/reports/accounts/host-report-eligibilty` is the problem, then I don't recommend commenting this code out as merging that backend PR I mentioned would solve the problem.

Comment (MomirMilutinovic) [inline src/app/http-interceptors/page-not-found.interceptor.ts]:
```suggestion
      catchError((error: HttpErrorResponse) => {
        if (error.status === 404) {
          this.router.navigate(['/404']);
        }
        return throwError(() => error)
      })
```

**Review by MomirMilutinovic — CHANGES_REQUESTED** [review]
I don't know if it's a good idea to comment out `PageNotFoundInterceptor`.
Also, I think it's cleaner to just remove it from the list of interceptors than to comment it out, since that makes it clearer that it's disabled.

File: src/app/http-interceptors/page-not-found.interceptor.ts
Comment (vradovic) [reply src/app/http-interceptors/page-not-found.interceptor.ts]: Resolved with merged backend PR.

**Review by MomirMilutinovic — APPROVED** [review]
Looks good.

**Pull request merged**
